src/ng/compile.js

```diff
@@ -3101,6 +3130,27 @@ function $CompileProvider($provide, $$sanitizeUriProvider) {
         }
       });
 
+      function recordChanges(key, currentValue, previousValue) {
+        if (isFunction(destination.$onChanges)) {
+          // If we have not already scheduled the onChanges hook then do so now
+          if (!changes) {
+            changes = {};
+            scope.$$postDigest(triggerOnChangesHook);
+          }
+          // Store this change
+          changes[key] = {previousValue: previousValue, currentValue: currentValue};
+        }
+      }
+
+      function triggerOnChangesHook() {
+        // We must run this hook in an apply since the $$postDigest runs outside apply
+        scope.$apply(function() {
```

Comment:
Yes you are right.
The alternative is to use `$applyAsync` but this has the downside of delaying the call to `$onChanges` by maybe 100ms.

What is better?